### PR TITLE
week-06-day-02-module-01

### PR DESCRIPTION
**Pull request by eerwitt**

### CODE REVIEW

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 1]
```diff
@@ -1 +1,445 @@
...
+
+1. First install the gem on your computer: `gem install awesome_print`
+2. Now just require it at the top of your file or in irb and then use it according to its documentation.
+  ```ruby
+  require 'awesome_print'
+  a = ["value1", "value2", "value3"]
+  ap a
+  ```
+
+[Ruby Toolbox](https://www.ruby-toolbox.com/)
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: Open in new tab. Also, what is this / why should they go there?

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 2]
```diff
@@ -1 +1,445 @@
...
+
+##Basic Challenges
+
+1. Make a folder called ruby-methods to keep all these ruby scripts we're about to write!
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: Inside `develop` folder, make a new folder...

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 3]
```diff
@@ -1 +1,445 @@
...
+
+1. Write a method called `get_contact` that
+  * takes in a `contacts` hash as a parameter, where the keys in `contacts` are names and the values are phone numbers,
+  * prompts the terminal for a **new** `name` and a `phone` number,
+  * and then adds the `name` and `phone` as a key value pair respectively **only if** `name` is not already a contact name,
+  * and `return`s the updated `contacts` hash.
+
+A sample contacts hash you could use to test would be `sample = {"Bob"=> 4152890045, "Emergency"=> 911, "Jenny"=> 8675309}`.
+
+1. Using `Array#map`, write a method called `get_responses` that takes an array of questions (strings) and returns an array of responses input from the console for each question. (Hint: you will need to use `gets.chomp` and `puts` ).
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: Markdown formatting gets weird here b/c the line above breaks the list. FYI.

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 4]
```diff
@@ -1 +1,445 @@
...
+
+**Challenge Set 3: Let's have a HTTParty!**
+
+4. Let's have an [HTTParty](https://github.com/jnunemaker/httparty)!
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: Open in new tab.

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 5]
```diff
@@ -1 +1,445 @@
...
+
+   <br><br><br><br><br>
+  * Hint: only check possible divisors up to half the original number
+   <br><br><br><br><br>
+  * Hint: only check possible divisors up to the square root of the original number
+    <br><br><br><br><br>
+  * Hint: if the number isn't 2, only check whether the number is divisible by odd numbers
+
+
+1. Find 3 gems you think are cool and try to use them in your own scripts. ([Ruby Toolbox](https://www.ruby-toolbox.com/)) has a lot of great gems listed, though you might have to hold off on incorporating Rails gems.
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: Open in new tab.

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 6]
```diff
@@ -1 +1,445 @@
...
+   The number is lower than 20.  Guess again!
+   17
+   The number is higher than 17.  Guess again!
+   18
+   The number is higher than 18.  Guess again!
+   19
+   You got 19 in 7 tries!
+   ```
+
+1. Rewrite `factorial` without recursion. For an extra challenge, write a recursive *and* an iterative version of a method that takes in a number `num` and calculate the `num`th [Fibionacci number](https://en.wikipedia.org/wiki/Fibonacci_number).
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: If you want them to do factorial AND fibonacci, maybe make this into two separate stretch challenges? Also, open link in new tab :)

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 7]
```diff
@@ -1 +1,445 @@
...
+  ```ruby
+  fibb(3)
+  => 2
+  ```
+
+1. Write a method to `reverse` a string [*in-place*](https://en.wikipedia.org/wiki/In-place_algorithm). You can only use a little extra space - think one extra character's worth.  Do not use iterators; use only a `while` loop and indexes.
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: Open in new tab.

File: week-06-intro-ruby-on-rails/day-02/module-01/readme.md [hunk 8]
```diff
@@ -1 +1,445 @@
...
+2. Now just require it at the top of your file or in irb and then use it according to its documentation.
+  ```ruby
+  require 'awesome_print'
+  a = ["value1", "value2", "value3"]
+  ap a
+  ```
+
+[Ruby Toolbox](https://www.ruby-toolbox.com/)
+
+##Basic Challenges
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/readme.md]: NOOOOOOO. I HATE "Basic" Challenges. :grimacing:  Sorry. I know, not your fault.

**Comment by eerwitt** [comment]
@cameronjacoby btw updated this with your feedback.

File: week-06-intro-ruby-on-rails/day-02/module-01/solutions.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+#     puts question
+#     response = gets.chomp
+#     responses.push(response)
+#   end
+#   responses
+# end
+
+
+
+# Stretch Challenges
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/solutions.md]: We're not doing these stretches anymore, right?

**Comment by cameronjacoby** [comment]
@eerwitt Just one comment in `solutions.md`, otherwise, looks awesome! Thanks for making all the changes!

File: week-06-intro-ruby-on-rails/day-02/module-01/solutions.md (hunk 9 above)
Comment (eerwitt) [inline week-06-intro-ruby-on-rails/day-02/module-01/solutions.md]: @cameronjacoby crud, I didn't see these. Updated.

Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-02/module-01/solutions.md]: @eerwitt Awesome, thanks!!

**Comment by cameronjacoby** [comment]
:+1:

**Pull request merged**